### DISCUSSION TITLE
Server does not start, using USB (UVC) camera module

### DISCUSSION THREAD

**Discussion by gusarg81** [bluenviron/mediamtx#662] [Questions]
## Which version are you using?

v0.17.6

## Which operating system are you using?

OS

- [x] Linux (Armbian 21.08.2 Focal)
- [ ] Windows
- [ ] macOS

Architecture

- [ ] amd64
- [x] arm64
- [ ] arm7
- [ ] arm6

## Description

When I run the server, with a local USB (UVC) camera, it does not start with the next:

Input #0, video4linux2,v4l2, from '/dev/video1':
  Duration: N/A, start: 70117.821835, bitrate: 165888 kb/s
    Stream #0:0: Video: rawvideo (YUY2 / 0x32595559), yuyv422, 1920x1080, 165888 kb/s, 5 fps, 5 tbr, 1000k tbn, 1000k tbc
2021/10/14 13:50:32 I [0/0] [RTSP] [conn [::1]:43118] opened
2021/10/14 13:50:32 I [0/0] [RTSP] [session 269661489] opened by [::1]:43118
2021/10/14 13:50:32 I [0/0] [RTSP] [conn [::1]:43118] ERR: invalid SDP: unable to get clock rate of track 0: attribute 'rtpmap' not found
2021/10/14 13:50:32 I [0/0] [RTSP] [conn [::1]:43118] closed
Could not write header for output file #0 (incorrect codec parameters ?): Server returned 400 Bad Request

## Config

> paths:
>   cam:
>     runOnInit: ffmpeg -f v4l2 -i /dev/video1 -c:v copy -preset ultrafast -tune zerolatency -b:v 600k -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
>     runOnInitRestart: yes

## Hardware

Rock64 (RK3328)

**Comment by aler9** [comment]
You can't copy the codec (`-c:v copy`) from the camera to the server - the camera sends raw video frames, the server accepts H264-encoded frames. You have to encode frames into H264:

```
 ffmpeg -f v4l2 -i /dev/video0 -c:v libx264 -preset ultrafast -tune zerolatency -b:v 600k -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
 ```

**Comment by gusarg81** [comment]
The problem is that SBC I have (Rock64), which has a RK3328 CPU (and running mainline kernel) still there is not way to encode using hwaccel (only decoding with hwaccel). So, using libx264 will lag a lot.

[a comment by an automated account is not shown]
